### PR TITLE
[Gateway API] - Distinguishing objects owning objects from addresses owning objects

### PR DESCRIPTION
**Pull request by patrickkuo**
This PR address issue stated in https://github.com/MystenLabs/sui/issues/1870#issuecomment-1125827018

Notable changes:
* replaced `get_owned_objects` by `get_object_owned_by_address` and `get_object_owned_by_object` in GatewayAPI and RPC API
* New `ObjectInfo` type contain info of an object, e.g. Type, owner, previous_tx
* return `ObjectInfo` instead of `ObjectRef` for the get_object_owned* methods
* use Owner instead of SuiAddress in `owner_index` 's key
* updates wallet cli `objects` command to display new `ObjectInfo`

New get_object_owned_by_* response
```
[
    {
        "jsonrpc": "2.0",
        "result": [
            {
                "objectId": "0x588c075e9cf063d0115bd4836edec6420b583237",
                "version": 0,
                "digest": "R0nQXC9fcnGH32PMd0G/makMYSGKpJGTAp0kOW+yWhQ=",
                "type": "0x2::Coin::Coin<0x2::SUI::SUI>",
                "owner": {
                    "AddressOwner": "0xabb44d0a462384b3453be5662cbb53f40046f317"
                },
                "previousTransaction": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
            },
            {
                "objectId": "0x5bd54483cdb466c573af1c89618d3ec3845630fa",
                "version": 0,
                "digest": "HR1hisi3QO1LvayEt7CrCEkrGh9wx22tOwenEx8AhJg=",
                "type": "0x2::Coin::Coin<0x2::SUI::SUI>",
                "owner": {
                    "AddressOwner": "0xabb44d0a462384b3453be5662cbb53f40046f317"
                },
                "previousTransaction": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
            },
            {
                "objectId": "0x85aace953774b56143f1f192d246fd037da5375a",
                "version": 0,
                "digest": "PVTTl3yjC+SJnnLsaDR6SWf2pP7XdmmcQVsRgh/ELPs=",
                "type": "0x2::Coin::Coin<0x2::SUI::SUI>",
                "owner": {
                    "AddressOwner": "0xabb44d0a462384b3453be5662cbb53f40046f317"
                },
                "previousTransaction": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
            },
            {
                "objectId": "0xaadd2cb4fce79e79ba5c5de428f395bd472c0743",
                "version": 0,
                "digest": "xyuNy04ZXhwZY3X63H5yPQCv6uaU3pznc0DDg5k//LQ=",
                "type": "0x2::Coin::Coin<0x2::SUI::SUI>",
                "owner": {
                    "AddressOwner": "0xabb44d0a462384b3453be5662cbb53f40046f317"
                },
                "previousTransaction": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
            },
            {
                "objectId": "0xaeaa3ad73d5843b63210bc49427306e9c6b99990",
                "version": 0,
                "digest": "k1xp+IcTKTqT4WKns59QSw7QcTYtxkgba43v50fuuyM=",
                "type": "0x2::Coin::Coin<0x2::SUI::SUI>",
                "owner": {
                    "AddressOwner": "0xabb44d0a462384b3453be5662cbb53f40046f317"
                },
                "previousTransaction": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
            }
        ],
        "id": 0
    }
]
```

### CODE REVIEW

**Comment by patrickkuo** [comment]
@666lcz this will need some changes in the SDK

File: sui/src/rpc_gateway.rs [hunk 1]
```diff
@@ -120,10 +120,17 @@ impl RpcGatewayServer for RpcGatewayImpl {
...
     }
 
-    async fn get_owned_objects(&self, owner: SuiAddress) -> RpcResult<ObjectResponse> {
-        debug!("get_objects : {}", owner);
-        let objects = self.gateway.get_owned_objects(owner).await?;
-        Ok(ObjectResponse { objects })
+    async fn get_objects_owned_by_address(
+        &self,
+        address: SuiAddress,
+    ) -> RpcResult<Vec<ObjectInfo>> {
```
Comment (666lcz) [inline sui/src/rpc_gateway.rs:126]: The return type `ObjectInfo` is a bit confusing with the existing `get_object_info` method. Can we either rename `ObjectInfo` to `ObjectSummary` or rename the `get_object_info` to `get_object`?

**Comment by bmwill** [comment]
@patrickkuo I've rebased the PR and you can find the result at https://github.com/bmwill/sui/tree/2087.

**Comment by patrickkuo** [comment]
> @patrickkuo I've rebased the PR and you can find the result at https://github.com/bmwill/sui/tree/2087.

Many thanks!

**Comment by patrickkuo** [comment]
Will leave this open for the TS SDK changes.

Thanks a lot @666lcz !!

**Pull request merged**